### PR TITLE
installed jest dependency and added plant unit tests

### PR DESCRIPTION
**Pull request by spencer-rafada**
- installed faker for creating fake data, @jest/globals, jest, ts-jest for using jest in TypeScript
- refactored getPlantById to findOne instead of find
- created tests folders for unit testing
- NOTE for plant.test.ts: 
* Line 13 of plant.test.ts, the unit test drops/deletes the collection after testing. We can choose to comment it out. Another option is to have a different MONGODB URI for DB TESTING so we don't mix production data and testing data.
* As of now, the test creates a plant model every test and that's because of the functionality of our controller functions. It doesn't really matter if we go and decide that we have a PRODUCTION DB and TEST DB. Just a suggestion.

Reference(s):
https://dev.to/dinckan_berat/testing-mongoose-with-ts-jest-264f

### CODE REVIEW

**Review by Jacksonckc — APPROVED** [review]
I am not familiar with Jest testing but everything else looks great! Great job lol.

**Review by jjossie — APPROVED** [review]
nice

**Review by spencer-rafada — COMMENTED** [review]
Unit tests for users look good.

**Review by jjossie — APPROVED** [review]
looks great 💪

File: src/routes/plant.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ plantRouter.get('/:plantId', async (req: Request, res: Response, next: NextFunct
   try {
     const id: Types.ObjectId = new Types.ObjectId(req.params.plantId);
     const result = await getPlantById(id);
-    if (result.length === 0) res.status(404).send({ message: `Plant not found` });
+    if (result === null) res.status(404).send({ message: `Plant not found` });
```
[the remark this answers is not in the document]
Comment (spencer-rafada) [reply src/routes/plant.ts]: I changed this already

**Pull request merged**